lumicks/pylake/nb_widgets/kymotracker_widgets.py

```diff
@@ -610,63 +574,114 @@ def __init__(
                     f"Lower bound should be lower than upper bound for parameter {key}"
                 )
 
-            self._slider_ranges[key] = slider_range
+        # update defaults to user-supplied parameters
+        arg_names, _, _, values = inspect.getargvalues(inspect.currentframe())
+        parameters = {name: values[name] for name in arg_names if name != "self"}
+        for key in keys:
+            if parameters[key] is not None:
+                algorithm_parameters[key].value = parameters[key]
+            if key in slider_ranges:
+                algorithm_parameters[key].lower_bound = slider_ranges[key][0]
+                algorithm_parameters[key].upper_bound = slider_ranges[key][1]
 
         super().__init__(
             kymo,
             channel,
             axis_aspect_ratio,
-            min_length,
             use_widgets,
             output_filename,
             algorithm,
             algorithm_parameters,
-            min_length_range=self._slider_ranges["min_length"],
             **kwargs,
         )
 
     def create_algorithm_sliders(self):
         import ipywidgets
 
-        window_slider = self._add_slider(
-            "Window",
-            "window",
-            "How many frames can a line disappear.",
-            *self._slider_ranges["window"],
-            slider_type=ipywidgets.IntSlider,
+        return ipywidgets.VBox(
+            [
+                self._add_slider(key, parameter)
+                for key, parameter in self._algorithm_parameters.items()
+                if parameter.ui_visible
+            ]
         )
-        thresh_slider = self._add_slider(
+
+
+@dataclass
+class KymotrackerParameter:
+    from numbers import Number
+
+    name: str
+    description: str
+    type: str
+    value: Number
+    lower_bound: Number
+    upper_bound: Number
+    ui_visible: bool
```

Comment:
I wonder if visible should raise if no bounds are given (since currently, we have no way of visualizing such sliders as we don't have spinbox support).